skorch/classifier.py

```diff
@@ -206,7 +206,7 @@ def predict(self, X):
         y_pred : numpy ndarray
 
         """
-        return super().predict_proba(X).argmax(axis=1)
+        return self.predict_proba(X).argmax(axis=1)
```

Comment:
I still find it a little strange that the generic `NeuralNet` has a `predict_proba`, but this has been around for a long time.

Should this be a part of a bug fix release?

Reply:
Yes, we can remove the method from `NeuralNet` (and `NeuralNetRegressor`), but it should come with a deprecation cycle. Do you want to take this?

To improve compatibility with sklearn even further, we should also look into [tags](https://scikit-learn.org/stable/developers/develop.html#estimator-tags), which partly come from the *mixins we currently don't use.

Reply:
> I will take it.

Great, thanks.

Regarding tags: Thanks for the reference. I took a look at the existing tags and to me it seems that some of them we can know in advance (e.g. `requires_fit`, `binary_only` for `NeuralNetBinaryClassifier`) and some we cannot (say, `multioutput`). So we could try to make a fine grained distinction between what estimators should support what tags, but don't even know all the places that they are being used. It might therefore not be worth the effort.